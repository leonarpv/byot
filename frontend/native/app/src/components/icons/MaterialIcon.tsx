import * as React from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {IconProps as BaseIconProps} from 'react-native-vector-icons/Icon';

export interface MaterialIconProps extends BaseIconProps {}

interface Props extends MaterialIconProps {}

export const MaterialIcon: React.FC<Props> = (props: Props) => {
  // this seems like it's unnecessary but when this is ready
  // https://github.com/oblador/react-native-vector-icons/issues/856
  // it'll be much more easier to add eg. `variant="outlined"` here
  // than in all its usages.
  return <Icon {...props} />;
};
