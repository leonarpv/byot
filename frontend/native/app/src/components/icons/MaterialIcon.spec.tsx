import * as React from 'react';
import {render} from '@testing-library/react-native';
import {MaterialIcon} from './MaterialIcon';

describe('<MaterialIcon/>', () => {
  it('should render', () => {
    const {asJSON} = render(<MaterialIcon name="keyboard-arrow-left" />);
    expect(asJSON()).toMatchSnapshot();
  });
});
