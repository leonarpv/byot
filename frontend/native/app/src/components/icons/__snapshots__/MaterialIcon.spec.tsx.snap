// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialIcon/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    îŒ”
  </Text>
</View>
`;
